### PR TITLE
2: add mail notification before locking user

### LINKED ISSUE #2: Add e-mail notification before locking user.
Add settings for e-mail notification before locking user.
It should be configuration parameter.

### PR DESCRIPTION
**Pull request by vifrrg**

### CODE REVIEW

File: oc_ldap_user_locker/locker.py [hunk 1]
```diff
@@ -196,24 +234,106 @@ def _process_single_user(self, ldap_c, user_dn):
...
+        _days_before_lock = self._get_days_before_lock(_lock_date)
+        logging.debug("Days before lock account '%s': %d" % (
+            _user_rec.get_attribute('cn'), _days_before_lock))
+
+        # check lock e-mail notifications
+        self._check_lock_notifications(
+                _user_rec, _conf, lock_date=_lock_date, days_before_lock=_days_before_lock)
+
+        if _days_before_lock > 0:
+            logging.debug("Is not the time to lock '%s', returning" % _user_rec.get_attribute('cn'))
```
Comment (ivansa-ru) [inline oc_ldap_user_locker/locker.py:256]: I may be picking, but I'd rather write 'returning None', or add a period. Looks like debug is missing, what specifically it is returning.

File: oc_ldap_user_locker/mailer.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+        return template_conf
+
+    def _get_smtp_client(self):
+        """
+        Return SMTP connection instance
+        """
+        _url = self._config.get("url")
+
+        if not re.match("(.*?:)?" + posixpath.sep + posixpath.sep, _url):
```
Comment (ivansa-ru) [inline oc_ldap_user_locker/mailer.py:89]: Maybe urllib.parse.urlpase.scheme?

File: oc_ldap_user_locker/locker.py (hunk 1 above)
Comment (vifrrg) [reply oc_ldap_user_locker/locker.py:256]: This method does not return a value at all.

File: oc_ldap_user_locker/mailer.py (hunk 2 above)
Comment (vifrrg) [reply oc_ldap_user_locker/mailer.py:89]: Unfortunately not.
This trick is implemented for `urlparse` to work correctly.
Example:
```
>>> urlparse.urlparse("bobby:1525/robert").scheme
'bobby'
```

**Review by ivansa-ru — APPROVED** [review]

**Pull request merged**